sparql-visualizer/visualizations/graphical/GraphicalVisualizerTest.spec.ts

```diff
@@ -0,0 +1,106 @@
+    }
+  }
+  union {
+  {
+    select ?work (min(?years) as ?year) (count(?coauthors) as ?number_of_authors) ?author_number where {
+      ?work (p:P50|p:P2093) ?author_statement .
+      ?author_statement ps:P50 wd:Q16733372 .
+      optional { ?author_statement pq:P1545 ?author_number . }
+      ?work (wdt:P50|wdt:P2093) ?coauthors .
+      ?work wdt:P577 ?dates .
+      bind(str(year(?dates)) as ?years) .
+    }
+    group by ?work ?author_number
+  }
+  bind(coalesce(if(?number_of_authors = 1,
+            'Solo author',
+            if(xsd:integer(?author_number) = 1,
+               'First author',
+               if(xsd:integer(?author_number) = ?number_of_authors,
+                  'Last author',
+                  'Middle author'))), 'Unknown')
+       as ?role)
+   }
+}
+group by ?year ?role
+order by ?year`;
+        const encodedURL: string =
+            "https://query.wikidata.org/embed.html#%23defaultView%3ABarChart%0A%23%20Inspired%20from%20LEGOLAS%20-%20http%3A%2F%2Fabel.lis.illinois.edu%2Flegolas%2F%0A%23%20Shubhanshu%20Mishra%2C%20Vetle%20Torvik%0Aselect%20%3Fyear%20(count(%3Fwork)%20as%20%3Fnumber_of_publications)%20%3Frole%20where%20%7B%0A%7B%0Aselect%20(str(%3Fyear_)%20as%20%3Fyear)%20(0%20as%20%3Fpages)%20(%22_%22%20as%20%3Frole)%20where%20%7B%0A%23%20default%20values%20%3D%200%0A%3Fyear_item%20wdt%3AP31%20wd%3AQ577%20.%0A%3Fyear_item%20wdt%3AP585%20%3Fdate%20.%0Abind(year(%3Fdate)%20as%20%3Fyear_)%0A%7B%0Aselect%20(min(%3Fyear_)%20as%20%3Fearliest_year)%20%20(max(%3Fyear_)%20as%20%3Flatest_year)%20where%20%7B%0A%3Fwork%20wdt%3AP50%20wd%3AQ16733372%20.%0A%3Fwork%20wdt%3AP577%20%3Fpublication_date%20.%0Abind(year(%3Fpublication_date)%20as%20%3Fyear_)%0A%7D%0A%7D%0Abind(year(now())%2B1%20as%20%3Fnext_year)%0Afilter%20(%3Fyear_%20%3E%3D%20%3Fearliest_year%20%26%26%20%3Fyear_%20%3C%3D%20%3Flatest_year)%0A%7D%0A%7D%0Aunion%20%7B%0A%7B%0Aselect%20%3Fwork%20(min(%3Fyears)%20as%20%3Fyear)%20(count(%3Fcoauthors)%20as%20%3Fnumber_of_authors)%20%3Fauthor_number%20where%20%7B%0A%3Fwork%20(p%3AP50%7Cp%3AP2093)%20%3Fauthor_statement%20.%0A%3Fauthor_statement%20ps%3AP50%20wd%3AQ16733372%20.%0Aoptional%20%7B%20%3Fauthor_statement%20pq%3AP1545%20%3Fauthor_number%20.%20%7D%0A%3Fwork%20(wdt%3AP50%7Cwdt%3AP2093)%20%3Fcoauthors%20.%0A%3Fwork%20wdt%3AP577%20%3Fdates%20.%0Abind(str(year(%3Fdates))%20as%20%3Fyears)%20.%0A%7D%0Agroup%20by%20%3Fwork%20%3Fauthor_number%0A%7D%0Abind(coalesce(if(%3Fnumber_of_authors%20%3D%201%2C%0A'Solo%20author'%2C%0Aif(xsd%3Ainteger(%3Fauthor_number)%20%3D%201%2C%0A'First%20author'%2C%0Aif(xsd%3Ainteger(%3Fauthor_number)%20%3D%20%3Fnumber_of_authors%2C%0A'Last%20author'%2C%0A'Middle%20author')))%2C%20'Unknown')%0Aas%20%3Frole)%0A%7D%0A%7D%0Agroup%20by%20%3Fyear%20%3Frole%0Aorder%20by%20%3Fyear";
+            expect(
+          constructGraphicalVisualizerUrl(sparqlQuery, endpoint, visualisationType)
+        ).toEqual(encodedURL);
+        /* tslint:enable: max-line-length */
```

Comment:
Statt für ganze Funktionen die Rule auszuschalten solltest du nur für die betroffenen Zeilen das tun.

Reply:
That's what I did, since the long line was within the query string, I couldn't only remove it for that specific line (`` string and all). In addition, the encodedURL and the test case itself were also too long. The test case problem is now solved by the formatting adjustments, so I can move the enabling comment up by a line.

Reply:
Wenn du die query zu etwas kurzem wechselst gibt es das Problem nicht

Reply:
Klar, ich habe die test cases nur so gemacht weil es der Originale Ansatz dafür war. Wir wollten ja auch schauen, dass wir die erstellten links verwenden/öffnen können. Da, wir es mit diesen Tests bewiesen haben kann ich sie, jetzt löschen und mit kürzeren ersetzen.

Reply:
Also am Ende sollten nur noch beim Encoder "echte" Queries sein, alle anderen haben damit nichts direkt zu tun. Also Ihnen ist egal was und wie escaped werden soll, die benutzen nur das Resultat des Encoders.

Reply:
Hmmm... ok. Jetzt bin ich wieder verwirrt... Weil mein Encoder ist die constructGraphicalVisualizerUrl(). Die Funktion encodiert die Query und setzt sie in die URL ein <- alles in einem. Es gibt nicht einen separaten encoder.  

Der Normalizer passt das Format der Query an, dort teste ich definitiv mit einer 'echten' query. Obwohl ich argumentieren würde, dass man den Normalizer auch mit kleineren strings besser testen kann. 

Somit testen wir am Ende nirgends, ob die Query korrekt encodiert wird, wenn ich diese Tests rausnehme.  Wollen wir das so?